File: aooc/basics/prime.java
import java.util.Scanner;

public class prime{
	public static void main(String [] args){
		int num=0;
		int cnt = 0;
		Scanner scen = new Scanner(System.in);
		System.out.print("Enter a number : ");
		num = scen.nextInt();
		if(num==1){
			System.out.println("Not odd nor even");
		}
		for(int i =2;i*i<=num;i++){
			if(num%i==0){
			       cnt++;
			}
	 	}
		if(cnt>0){
			System.out.println("Not prime");
		}
		else{
			System.out.println("Prime");
		}
		scen.close();
	}
}

File: Practical/aooc/basics/cpp/prime.cpp
#include<iostream>
using namespace std;

int main(){
    int num;
    cout<<"Enter a number :";
    cin>>num;
    bool Flag = false;
    if(num==1){
        cout<<"Nor prime nor composite";
    }

    for (int i=2;i*i<=num;i++){
        if(Flag){
            cout<<"Num is not prime";
            break;
        }
        if(num%i==0){
            Flag = true;
        }
    }
    if(!Flag){
        cout<<"Number is prime";
    }
    
    return 0;
}